provider/resource_rediscloud_acl_user.go

```diff
@@ -0,0 +1,172 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/RedisLabs/rediscloud-go-api/redis"
+	"github.com/RedisLabs/rediscloud-go-api/service/access_control_lists/users"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"strconv"
+	"time"
+)
+
+func resourceRedisCloudAclUser() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Create an ACL User within your Redis Enterprise Cloud Account",
+		CreateContext: resourceRedisCloudAclUserCreate,
+		ReadContext:   resourceRedisCloudAclUserRead,
+		UpdateContext: resourceRedisCloudAclUserUpdate,
+		DeleteContext: resourceRedisCloudAclUserDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Read:   schema.DefaultTimeout(3 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "A meaningful name to identify the user",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"role": {
+				Description: "The role which the user has",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"password": {
+				Description: "The user's password",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func resourceRedisCloudAclUserCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*apiClient)
+	var diags diag.Diagnostics
+
+	name := d.Get("name").(string)
+	role := d.Get("role").(string)
+	password := d.Get("password").(string)
+
+	createUser := users.CreateUserRequest{
+		Name:     redis.String(name),
+		Role:     redis.String(role),
+		Password: redis.String(password),
+	}
+
+	id, err := api.client.Users.Create(ctx, createUser)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(strconv.Itoa(id))
+
+	return diags
```

Comment:
It's common to chain the read function up to create (& update) to ensure the correct values have been read back from the server. It's also not uncommon to call the update function at the end of the create function, especially when certain attributes can only be set _after_ a resource exists.

Reply:
For the first issue (calling read after create/update), we've never seen an example where the correct values were not set. Relative to subscriptions and databases, the ACL entites are quite simple and work consistently. Would you still recommend a call to read afterwards?

As for the second issue, there are no properties that can't be set at create-time. Due to the simplicity of the resources, there's no need to update _after_ creation in these cases. Unless I'm missing the point and you're looking at something I should be updating?

Reply:
I would still perform `read` at the end of `create`/`update`.

On whether `create` should call `update` - it kind of depends on the complexity of the resource being implemented. You'll see most resources in HashiCorp-maintained providers, like terraform-provider-aws, do this - but you are correct that the resource isn't sufficiently complex to need to worry about it.